P1/Project_Code/frontend/src/routes/SearchResults/SearchResults.route.jsx

```diff
@@ -8,15 +8,27 @@ export const SearchResults = () => {
     // Grab the filtered movies
     const { filteredMovies } = useContext(MoviesContext);
 
-    return (
-        <Grid container spacing={{ sm: 2, md: 3 }} columns={{ xs: 12, sm: 4, md: 3 }}>
-            {
-                filteredMovies.map((movie, index) => (
-                    <Grid item key={index}>
-                        <MediaCard movie={movie} type="searched"/>
-                    </Grid>
-                ))
-            }
-        </Grid>
-    )
+    // Show movies searched in case there were any
+    if (filteredMovies.length > 0)
```

Comment:
Suggest refactoring this code to include a check that filteredMovies is not null or undefined . See example below 
```
// Check if filteredMovies is defined and not null
 if (filteredMovies && filteredMovies.length > 0) {
some code here
}
// else if no movies were filtered or filteredMovies is undefined/null, display "No Results Found" message
else {
some code here
}
```

Reply:
The condition `if (filteredMovies && filteredMovies.length > 0)` checks if `filteredMovies` is not `undefined` or `null` before checking its `length`. This prevents potential runtime errors when trying to access the length property of an undefined or null value.